cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/TransactionServiceManager.java

```diff
@@ -56,7 +58,7 @@ public boolean isServiceAvailable() {
     try {
       Discoverable discoverable = new RandomEndpointStrategy(() -> discoveryServiceClient.discover(serviceName))
         .pick(discoveryTimeout, TimeUnit.SECONDS);
-      if (discoverable == null) {
+      if (discoverable == null && !isSql) {
```

Comment:
cConf is a protected variable in abstract class. Can we use that directly rather than introducing a new boolean variable?

Reply:
I added the boolean so that we don't need to fetch from cConf and compare each time the method is called. The system service status call is the most frequent call in MonitorHandler I believe

Reply:
May be call it `isSqlStorageImpl` or something similar?